pkg/scheduler/scheduler.go

```diff
@@ -800,9 +815,25 @@ func (p *podPreemptorImpl) deletePod(pod *v1.Pod) error {
 }
 
 func (p *podPreemptorImpl) setNominatedNodeName(pod *v1.Pod, nominatedNodeName string) error {
+	klog.V(3).Infof("Setting nominated node name for %s/%s to \"%s\"", pod.Namespace, pod.Name, nominatedNodeName)
 	podCopy := pod.DeepCopy()
+	if podCopy.Status.NominatedNodeName == nominatedNodeName {
+		return nil
+	}
+	oldData, err := json.Marshal(podCopy)
+	if err != nil {
+		return err
+	}
 	podCopy.Status.NominatedNodeName = nominatedNodeName
-	_, err := p.Client.CoreV1().Pods(pod.Namespace).UpdateStatus(context.TODO(), podCopy, metav1.UpdateOptions{})
+	newData, err := json.Marshal(podCopy)
+	if err != nil {
+		return err
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, &v1.Pod{})
+	if err != nil {
+		return fmt.Errorf("failed to CreateTwoWayMergePatch for pod %q/%q: %v", pod.Namespace, pod.Name, err)
+	}
+	_, err = p.Client.CoreV1().Pods(pod.Namespace).Patch(context.TODO(), pod.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
 	return err
```

Comment:
@Huang-Wei  in the original code we returned an error in case the status update failed (i.e., the case that we hit in the test), which means the scheduling cycle of the the high priority pod failed in the test, and later got scheduled again and either evicted the medium pod if there were no enough resources, or simply coexisted with medium pod? This seems to me a reasonable behavior, and that the problem is with the test itself (asserting that NominatedNodeName field getting cleared), not the code, what do you think?

Reply:
well, the high priority pod is already failed to schedule, that is why we are here in the first place, alas I don't think we are fixing a bug here, I think the test is making a strict assumption, and so, unless the new code is not more expensive than the original one, we should modify the test not the code.

Reply:
See my comment https://github.com/kubernetes/kubernetes/pull/90978#issuecomment-627153537.